plugins/scaffolder/src/components/ScaffolderPage/ScaffolderPage.tsx

```diff
@@ -85,10 +89,12 @@ export const ScaffolderPageContents = ({
       />
       <Content>
         <ContentHeader title="Available Templates">
-          <CreateButton
-            title="Register Existing Component"
-            to={registerComponentLink && registerComponentLink()}
-          />
+          {allowed && (
```

Comment:
This and the other PR got me thinking about the UX here. If you permission check takes some time, we'll have a bit of flickering going on.

A possible solution here could be to cache permission queries? Frontend only ofc, and we'd still do the request every time. Just to use it to guess the response so that we can render the correct thing from the beginning most of the time. Logic would basically be that if it was allowed last time, then default `allowed` to `true` during loading. `loading` would still be set as expected and if you need to know the exact permission before continuing you would wait until `loading` is `false`, for example for entity pages.

Reply:
I like that idea! I'll break this out into a separate PR.

Reply:
#9064